instance/models/load_balancer.py

```diff
@@ -111,6 +145,55 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.domain
 
+    def modifies_configuration(method):  # nopep8 # pylint: disable=no-self-argument
+        """
+        Flag an internal method as one that modifies the state of load balancer configuration,
+        to allow managing configuration state outside and around the method call.
+        """
+        @functools.wraps(method)
+        def wrapper(self, *args, **kwargs):
+            """
+            Used to wrap the function that modified load balancer configuration state.
+            """
+            try:
+                # Prevent race conditions by updating state only if the LB is actually clean or dirty.
+                updated = LoadBalancingServer.objects.filter(
+                    pk=self.pk,
+                    configuration_state__in=[self.CLEAN, self.DIRTY]
+                ).update(
+                    configuration_state=self.IN_PROGRESS
+                )
```

Comment:
It does come with a single command, yes, otherwise this would be totally useless for the purpose of locking.

Yeah, I think we can refresh from the DB if this does succeed (and there wouldn't be a race condition there because only a single thread will be able to succeed at a time here). Or we can just replace `self.configuration_state` with querysets.

Reply:
Clarification on `update`:

From https://docs.djangoproject.com/en/1.11/ref/models/querysets/#django.db.models.query.QuerySet.update --

> Using update() also prevents a race condition wherein something might change in your database in the short period of time between loading the object and calling save().

Reply:
I also tested locally and got:

```sql
UPDATE "instance_loadbalancingserver"
SET "configuration_state" = 'in_progress'
WHERE ("instance_loadbalancingserver"."configuration_state" IN ('dirty',
                                                                'clean')
       AND "instance_loadbalancingserver"."id" = 2)

```

Reply:
I also believed that it results in a single `UPDATE`, but I wanted to know for sure.  Thanks!